llama.h

```diff
@@ -195,15 +195,17 @@ extern "C" {
         LLAMA_KV_OVERRIDE_TYPE_INT,
         LLAMA_KV_OVERRIDE_TYPE_FLOAT,
         LLAMA_KV_OVERRIDE_TYPE_BOOL,
+        LLAMA_KV_OVERRIDE_TYPE_STR,
     };
 
     struct llama_model_kv_override {
         char key[128];
         enum llama_model_kv_override_type tag;
         union {
             int64_t int_value;
-            double float_value;
-            bool bool_value;
+            double  float_value;
+            bool    bool_value;
+            char *  str_value;
```

Comment:
I think it will be simpler if we keep the string values on the stack:

```c
char key[128];
char value[128]; <-- here
```

The way it is now seems too error-prone - can easily forget to free the memory.

Maybe also the `union` has to go away and parse the string value lazily when needed. But we can do this separately